### PR TITLE
block_resize: add support for data plane

### PR DESCRIPTION
**Pull request by aliang123**
block_resize: add support for data plane

Signed-off-by:Aihua Liang<aliang@redhat.com>

Bug id:1434266

### CODE REVIEW

File: qemu/tests/block_resize.py [hunk 1]
```diff
@@ -73,7 +73,7 @@ def compare_block_size(session, block_cmd, block_pattern):
 
     drive_path = ""
     if params.get("os_type") == 'linux':
-        drive_id = params["blk_extra_params_%s" % data_image].split("=")[1]
+        drive_id = params["blk_extra_params_%s" % data_image].split(",")[0].split("=")[1]
```
Comment (suqinhuang) [inline qemu/tests/block_resize.py]: different style for blk_extra_params_stg, you can use function to define it

blk_extra_params_stg = "serial=TARGET_DISK0"
blk_extra_params_stg = "wwn=0x5000123456789abc"

File: qemu/tests/block_resize.py [hunk 2]
```diff
@@ -73,7 +73,12 @@ def compare_block_size(session, block_cmd, block_pattern):
 
     drive_path = ""
     if params.get("os_type") == 'linux':
-        drive_id = params["blk_extra_params_%s" % data_image].split("=")[1]
+        pattern = r"(serial|wwn)=(\w+)"
+        match = re.search(pattern, params["blk_extra_params_%s" % data_image], re.M)
+        if match:
+            drive_id = match.group(2)
+        else:
+            raise test.fail("No available tag to get drive id")
```
Comment (suqinhuang) [inline qemu/tests/block_resize.py]: no raise, use test.fail directly

File: qemu/tests/cfg/block_resize.cfg [hunk 3]
```diff
@@ -59,3 +59,14 @@
...
+        - @default:
+        - data_plane:
+            only Host_RHEL.m7
+            only virtio_blk virtio_scsi
+            iothreads = iothread0
+            virtio_blk:
+                blk_extra_params_stg += ",iothread=${iothreads}"
+            virtio_scsi:
+                no Host_RHEL.m7.u0, Host_RHEL.m7.u1, Host_RHEL.m7.u2
+                bus_extra_params_stg = "iothread=${iothreads}"
```
Comment (suqinhuang) [inline qemu/tests/cfg/block_resize.cfg]: how to get drive_id for virtio_scsi as you didn't transfer wwn|serial

File: qemu/tests/cfg/block_resize.cfg [hunk 4]
```diff
@@ -59,3 +59,14 @@
         disk_change_ratio = 1.5 0.5
     RHEL.5:
         need_reboot = yes
+    variants:
+        - @default:
+        - data_plane:
```
Comment (suqinhuang) [inline qemu/tests/cfg/block_resize.cfg]: block_resize is a basic test, make sure it doesn't effect other test when add variants

File: qemu/tests/block_resize.py (hunk 2 above)
Comment (aliang123) [reply qemu/tests/block_resize.py]: thanks, will update it.

Comment (suqinhuang) [reply qemu/tests/block_resize.py]: for line 9 image_snapshot_image1 = yes, as I know, snapshot don't support resize

File: qemu/tests/cfg/block_resize.cfg (hunk 3 above)
Comment (aliang123) [reply qemu/tests/cfg/block_resize.cfg]: for virtio-scsi, "iothread=${iothreads}" is added to virtio-scsi-pci, no effect to blk params.

File: qemu/tests/cfg/block_resize.cfg (hunk 4 above)
Comment (aliang123) [reply qemu/tests/cfg/block_resize.cfg]: It just effects internal test loop "virtio-win-acceptance.cfg" and after discussion, it's acceptable.

**Comment by peixiu** [comment]
Tried to test this case on RHEL7.4 and Win2008-64&R2. Passed on RHEL7.4 and Win2008-R2 guests, on Win2008-64, it failed with error. The results as follows:

RHEL7.4:
 (3/4) smp_2.8192m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_blk.up.virtio_net.RHEL.7.4.x86_64.io-github-autotest-qemu.block_resize: PASS (477.22 s)
 (4/4) smp_2.8192m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_blk.up.virtio_net.RHEL.7.4.x86_64.io-github-autotest-qemu.block_resize.data_plane: PASS (97.05 s)

Win2008-64&R2:
 (3/6) smp_2.8192m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_blk.up.virtio_net.Win2008.x86_64.sp2.io-github-autotest-qemu.block_resize: ERROR (165.00 s)
 (4/6) smp_2.8192m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_blk.up.virtio_net.Win2008.x86_64.sp2.io-github-autotest-qemu.block_resize.data_plane: ERROR (164.50 s)
 (5/6) smp_2.8192m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_blk.up.virtio_net.Win2008.x86_64.r2.io-github-autotest-qemu.block_resize: PASS (150.17 s)
 (6/6) smp_2.8192m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_blk.up.virtio_net.Win2008.x86_64.r2.io-github-autotest-qemu.block_resize.data_plane: PASS (165.76 s)

File: qemu/tests/cfg/block_resize.cfg (hunk 3 above)
Comment (CongLi) [reply qemu/tests/cfg/block_resize.cfg]: @suqinhuang  wwn/serial was defined in line 13 and 15 in block_resize.cfg

**Comment by CongLi** [comment]
Acked-by: Cong Li <coli@redhat.com>

File: qemu/tests/block_resize.py [hunk 5]
```diff
@@ -73,7 +73,12 @@ def compare_block_size(session, block_cmd, block_pattern):
 
     drive_path = ""
     if params.get("os_type") == 'linux':
-        drive_id = params["blk_extra_params_%s" % data_image].split("=")[1]
+        pattern = r"(serial|wwn)=(\w+)"
+        match = re.search(pattern, params["blk_extra_params_%s" % data_image], re.M)
```
Comment (hereischen) [inline qemu/tests/block_resize.py]: line too long 84 characters

Comment (aliang123) [reply qemu/tests/block_resize.py]: @hereischen , have updated, thanks for your reviewing.

**Comment by hereischen** [comment]
Acked-by: Haotong Chen hachen@redhat.com

**Comment by CongLi** [comment]
Thanks all.

**Pull request merged**